### ISSUE TITLE
Problem with FrankenPhpSymfony Runtime running container in production mode

### ISSUE THREAD

**Issue by michelcourtade** [dunglas/symfony-docker#506]
Hi all, 

thanks for the project, it works like a charm in dev mode and my project is running perfectly on my computer.

I try to install it on our servers and run it in prod mode but I have an error running the container. Due to Let's Encrypt limitations, I can't run it with a server name because my domain is not reachable on the internet (only on our intranet).

I launch my containers with this command :

`APP_ENV=prod SERVER_NAME=:80 APP_SECRET=mysecret CADDY_MERCURE_JWT_SECRET=caddysecret POSTGRES_DB=mydb POSTGRES_USER=myuser  POSTGRES_PASSWORD=mypass docker compose -f compose.yaml -f compose.prod.yaml up --wait`

After seconds waiting for frankenphp to launch, docker responds `container my_app-php-1 is unhealthy`.

And during the launch, if I run a `docker logs mycontainerid` I have these errors :

`{"level":"error","ts":1699521868.004604,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.0059345,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.0063403,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0069566,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0077014,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.008419,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0096755,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.0103762,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0127456,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.014324,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0155506,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.0166526,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.017027,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0189488,"msg":"unexpected termination, restarting","worker":"/app/public/index.php"}
{"level":"error","ts":1699521868.0198276,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
{"level":"error","ts":1699521868.0245843,"msg":"PHP Fatal error:  Uncaught Error: Class \"Runtime\\FrankenPhpSymfony\\Runtime\" not found in /app/vendor/autoload_runtime.php:16\nStack trace:\n#0 /app/public/index.php(5): require_once()\n#1 {main}\n  thrown in /app/vendor/autoload_runtime.php on line 16","syslog_level":"err"}
`

The build seems correct, here is the output of the build command : 

`[+] Building 142.1s (26/26) FINISHED                                                                                                                                                                         docker:default
 => [php internal] load build definition from Dockerfile                                                                                                                                                               
 => => transferring dockerfile: 2.81kB                                                                                                                                                                                 
 => [php internal] load .dockerignore                                                                                                                                                                                  
 => => transferring context: 498B                                                                                                                                                                                      
 => [php] resolve image config for docker.io/docker/dockerfile:1.4                                                                                                                                                     
 => CACHED [php] docker-image://docker.io/docker/dockerfile:1.4@sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc                                                                                
 => [php internal] load metadata for docker.io/composer/composer:2-bin                                                                                                                                                 
 => [php internal] load metadata for docker.io/dunglas/frankenphp:latest-alpine                                                                                                                                        
 => [php frankenphp_upstream 1/1] FROM docker.io/dunglas/frankenphp:latest-alpine@sha256:a1516071102abb4e3630fa95f347a7d9c95de1a3771220e8661822ea75dd70c2                                                              
 => [php internal] load build context                                                                                                                                                                                  
 => => transferring context: 7.48kB                                                                                                                                                                                    
 => CACHED [php composer_upstream 1/1] FROM docker.io/composer/composer:2-bin@sha256:abecd5d448bf7c3cc03bc07ed2920e939a5d7a75f821a58455458e25014c9297                                                                  
 => CACHED [php frankenphp_base 1/8] WORKDIR /app                                                                                                                                                                      
 => [php frankenphp_base 2/8] RUN apk add --no-cache   acl   file   gettext   git  ;                                                                                                                                   
 => [php frankenphp_base 3/8] RUN set -eux;  install-php-extensions   apcu   intl   opcache   zip  ;                                                                                                                  
 => [php frankenphp_base 4/8] RUN apk add --no-cache --virtual .pgsql-deps postgresql-dev;  docker-php-ext-install -j"$(nproc)" pdo_pgsql;  apk add --no-cache --virtual .pgsql-rundeps so:libpq.so.5;  apk del .pgs  
 => [php frankenphp_base 5/8] COPY --link frankenphp/conf.d/app.ini /usr/local/etc/php/conf.d/                                                                                                                         
 => [php frankenphp_base 6/8] COPY --link --chmod=755 frankenphp/docker-entrypoint.sh /usr/local/bin/docker-entrypoint                                                                                                 
 => [php frankenphp_base 7/8] COPY --link frankenphp/Caddyfile /etc/caddy/Caddyfile                                                                                                                                    
 => [php frankenphp_base 8/8] COPY --from=composer_upstream --link /composer /usr/bin/composer                                                                                                                         
 => [php frankenphp_prod 1/8] RUN mv "/usr/local/etc/php/php.ini-production" "/usr/local/etc/php/php.ini"                                                                                                              
 => [php frankenphp_prod 2/8] COPY --link frankenphp/conf.d/app.prod.ini /usr/local/etc/php/conf.d/                                                                                                                    
 => [php frankenphp_prod 3/8] COPY --link frankenphp/worker.Caddyfile /etc/caddy/worker.Caddyfile                                                                                                                      
 => [php frankenphp_prod 4/8] COPY --link composer.* symfony.* ./                                                                                                                                                      
 => [php frankenphp_prod 5/8] RUN set -eux;  composer install --no-cache --prefer-dist --no-dev --no-autoloader --no-scripts --no-progress                                                                             
 => [php frankenphp_prod 6/8] COPY --link . ./                                                                                                                                                                         
 => [php frankenphp_prod 7/8] RUN rm -Rf frankenphp/                                                                                                                                                                   
 => [php frankenphp_prod 8/8] RUN set -eux;  mkdir -p var/cache var/log;  composer dump-autoload --classmap-authoritative --no-dev;  composer dump-env prod;  composer run-script --no-dev post-install-cmd;  chmod +  
 => [php] exporting to image                                                                                                                                                                                           
 => => exporting layers                                                                                                                                                                                                
 => => writing image sha256:649525623067e6fe1167b3164fa66c8bf2046b10cd3f7f8250b8c60b1e05f60e                                                                                                                           
 => => naming to docker.io/library/app-php   `

After the failed "up" command, I can connect to the container and if I launch a bin/console command (like bin/console cache:clear) it works perfectly. But the container is restarting every 30 sec (automaticly trying to up the container ?) 

What do you think I can do to investigate this problem ? For me, building without returning an error, means everything is correctly installed, especially the FrankenPhpSymfony Runtime ?

Thanks for your help !

**Comment by 11k** [comment]
Are you positive the FrankenPHP Symfony Runtime is installed?

Coincidentally, I just ran into the same exact issue you did. In my case, I added Symfony Docker to an existing project. I believe the instructions for doing so don't mention that you have to install the FrankenPHP Runtime, but it's required in the production FrankenPHP image. If you create a new project using Symfony Docker, it automatically adds the runtime to your dependencies [here](https://github.com/dunglas/symfony-docker/blob/main/frankenphp/docker-entrypoint.sh#L16).

**Comment by dunglas** [comment]
Yes we need to explain that this should be installed, or that the worker mode should be disabled (which is probably safer for existing projects anyway).

**Comment by michelcourtade** [comment]
Ok thank you @11k for your reply, it seems to be that runtime/frankenphp-symfony was not installed as you said. (I also added Symfony Docker to an existing symfony application)
@dunglas I'll try to update the doc but disabling the worker mode should be better. I don't know if it's a Caddy feature but I did'nt see any mention of that in the docs. Did I miss something ?

It's now working perfectly, and I'm trying now to deploy assets with Encore on the prod container. Even if the build dir is on the host, the directory is not copied on the public/ dir of the frankenphp container. I suppose it's normal but I try to find the correct way to do it.

Thanks again for the package, it rocks !
